addons/website/static/src/js/menu/content.js

```diff
@@ -224,10 +226,28 @@ var PagePropertiesDialog = weWidgets.Dialog.extend({
             redirect_type: this.$('#redirect_type').val(),
             website_indexed: this.$('#is_indexed').prop('checked'),
             visibility: this.$('#visibility').val(),
-            visibility_password: this.$('#visibility').val() === 'password' ? this.$('#visibility_password').val() : false,
             visibility_group: this.$('#visibility').val() === 'restricted_group' ? this.$('#visibility_group').data('group-id') : false,
             date_publish: datePublish,
         };
+        if (this.$('#visibility').val() === 'password') {
+            params['visibility_pwd'] = this.$('#visibility_password').val();
+
+            if (!params['visibility_pwd']) {
+                this.current_modal = this.$el.parents('.modal')
+                this.current_modal.hide();
+                Dialog.alert(
+                    null,
+                    _t("Password is required"),
+                    {
+                        confirm_callback: function() {
+                            self.current_modal.show();
+                        }
+                    }
+                );
+                return;
+            }
+        }
```

Comment:
Possible alternative (don't know the details): set the `required` attribute on the `#visibility_password` based on the value of `#visibility`, then call `checkValidity()` (on `#visibility_password`). That should display a relevant error message without having to swap modals and returns `true` if valid and `false` otherwise.